Lib/idlelib/percolator.py

```diff
@@ -38,6 +38,21 @@ def insertfilter(self, filter):
         filter.setdelegate(self.top)
         self.top = filter
 
+    def insertfilterafter(self, filter, after):
+        assert isinstance(filter, Delegator)
+        assert isinstance(after, Delegator)
+        assert filter.delegate is None
+
+        f = self.top
+        f.resetcache()
+        while f is not after:
```

Comment:
Assert strikes me as an odd (confusing) way to end a loop.  More normal:
```suggestion
        while f is not after and f is not self.bottom:
```
followed after the loop by
        if f is self.bottom: raise ValueError("after must be in the delegator chain")

Reply:
Note that this mirrors the very similar piece of logic in `Percolator.removefilter()`.

The assert isn't ending the loop. The code is kept simpler by assuming that `after` is in the percolator.

I agree, more generally, that Percolator should be made safer and better tested, or removed entirely in favor of refactoring to a less confusing design. However, given its current state, I believe that this change is in line with the "status quo".

Reply:
It is hard to review diffs to code that needs an overhaul.  So leave what you did until then.